int8 flag
int8 TZ
